### PR TITLE
DR-798 Documentation Update - Developer Onboarding

### PR DESCRIPTION
**Pull request by Aish1990**

### CODE REVIEW

File: docs/components/LiveData.vue [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    <p><strong>Read time:</strong> {{ readTime }}</p>
+    <p><strong>Views:</strong> {{ views }}</p>
+  </div>
+</template>
+
+<script>
+export default {
+  data() {
+  }
+}
```
Comment (dmohns) [inline docs/components/LiveData.vue]: Please remove this. It's not really needed.

File: docs/dev_docs/developer-onboarding.md [hunk 2]
```diff
@@ -3,3 +3,245 @@ order: 2
 ---
 
 # Developer Onboarding
+
+<LiveData />
+
+**<span style="font-size: 12px;">Owned by Vijay (Unlicensed)</span>**<br>
+**<span style="font-size: 12px;">Last updated: May 14, 2024 by Daniel</span>**<br>
+**<span style="font-size: 12px;">6 min read</span>**<br>
+**<span style="font-size: 12px;">18 people viewed</span>**
```
Comment (dmohns) [inline docs/dev_docs/developer-onboarding.md]: Please remove all this, not needed (and wrong)

File: docs/dev_docs/developer-onboarding.md [hunk 3]
```diff
@@ -3,3 +3,245 @@ order: 2
...
+
+On the successful registration of the organisation, organisation admin user will see registered organisation details.
+
+“Add Project” button appears on Organisation Details page.
+
+## <u>Project Registration</u>
+
+This is new API development from D-Rec platform. User will enter the following details while registering project.
+
+NOTE - <span style="color: red;">Ricky will confirm us on considering I-REC standards to be used while creating D-Rec certificate</span>
```
Comment (dmohns) [inline docs/dev_docs/developer-onboarding.md]: Please update this comment. Remove Ricky's name from here. But add a note where the I-REC standards are tracked.

File: docs/dev_docs/developer-onboarding.md [hunk 4]
```diff
@@ -3,3 +3,245 @@ order: 2
...
+    Status of the project (dropdown) - Current
+    Technology Type (dropdown) - Solar Lanterns, Solar Home Systems, Community Mini Grids, Others
+    On selection of Others, text field should appear
+    Aggregate Nameplate Capacity for project (kW)
+    Expected or Actual kWhs to be generated from the first year of operation
+    Description of the project
+    Electricity off-taker category (dropdown) - Agriculture, Manufacturing, Public Services etc…
+    Sector
+    Description of Social / Environmental Impact
+    Project Image to browse (types accepted - PNGs, JPEGs)
```
Comment (dmohns) [inline docs/dev_docs/developer-onboarding.md]: This is formatted as a code block. Is this correct?

File: docs/dev_docs/developer-onboarding.md [hunk 5]
```diff
@@ -3,3 +3,245 @@ order: 2
...
+Grid Interconnection (dropdown) - True , False<br>
+Commissioning / Commissioned Year (dropdown) - 2021, 2022 etc…<br>
+Quarter of the Year (dropdown) - Q1, Q2, Q3 and Q4<br>
+Nameplate Capacity for Site (kW)<br>
+Expected Year One kWhs<br>
+Site Image Upload (expected image formats - PNGs, JPEGs)<br>
+
+This registration page will consist of Submit (creating new entry), Reset (for resetting the page) and Back (to disappear form and go back to project details page). 
+
+NOTE - If user selects Single Site entry option, then site registration form will appear. Otherwise, form with file browse field and submit button will appear.
```
Comment (dmohns) [inline docs/dev_docs/developer-onboarding.md]: Please use a `NOTE` Admonition here: https://vitepress.dev/guide/markdown#github-flavored-alerts

**Review by dmohns — COMMENTED** [review]
Overall looks good. Left a few comments..

Please also, remove all the HTML tags from the Markdown, like `&nbsp;`, `<span style="background-color: orange">`, `<u>`, etc...

**Review by dmohns — COMMENTED** [review]
One more comment, please move all the images into `img` subfolder of `dev_docs`

File: docs/.vitepress/config.mjs [hunk 6]
```diff
@@ -5,7 +5,7 @@ import { withMermaid } from 'vitepress-plugin-mermaid';
 // https://vitepress-sidebar.jooy2.com/getting-started
 const vitepressSidebarOptions = {
   excludeFiles: 'README.md',
-  excludeFilesByFrontmatter: true,
+  excludeFilesByFrontmatterFieldName: true,
```
Comment (dmohns) [inline docs/.vitepress/config.mjs]: Please don't do this, or at least explain why you think it's required?

File: docs/dev_docs/developer-onboarding.md [hunk 7]
```diff
@@ -3,3 +3,343 @@ order: 2
...
+- Last Name
+- Organisation Name
+- Organisation Email Address
+- Telephone
+- User Name (for login) as we will use email address as username
+- Password
+- Confirm Password
+- Organisation Type (radio buttons) - Developer, Buyer
+
+![User Registration Flow](img/developer-onboarding/fef5c2ea-9e92-4316-9fe9-e45eff0a327d.png)
```
Comment (dmohns) [inline docs/dev_docs/developer-onboarding.md]: Please convert this into a Markdown Chart (Flowchart) as well.

File: docs/dev_docs/developer-onboarding.md [hunk 8]
```diff
@@ -3,3 +3,343 @@ order: 2
...
+## Organisation Registration (Developer)
+
+### Organisation Details Page
+
+Organisation admin user will successfully login to D-Rec platform and see “No organisations registered” message and button to register organisation.
+
+> [!NOTE]
+> This message is subjective to change as per future requirements.
+
+![Organization regitration flow](img/developer-onboarding/99fb299e-ab54-4524-a08d-4b9c70e05ca1.png)
```
Comment (dmohns) [inline docs/dev_docs/developer-onboarding.md]: Please convert this into a Markdown Chart (Flowchart) as well.

File: docs/dev_docs/developer-onboarding.md [hunk 9]
```diff
@@ -3,3 +3,343 @@ order: 2
...
+
+Upon successful registration of project, project details should appear in tabular format along with search (filter) field on project details page. This filter should appear on tabular data.
+
+Each row in table should consist of “Details” button and “Add Site” button. These details should appear on Organisation Details page.
+
+## Device Onboarding (Site Registration)
+
+User fills the following details on Site Registration Page. Site Registration is happening against to the project. Hence, Site Registration Page will auto populate Project Name, Off Taker, Sector and Impact Story details.
+
+![Device Onboarding Flow](img/developer-onboarding/6faf99ae-11e2-4b95-89d8-c027cc2542bf.png)
```
Comment (dmohns) [inline docs/dev_docs/developer-onboarding.md]: Please convert this into a Markdown Chart (Flowchart) as well.

**Review by dmohns — COMMENTED** [review]
Left some comments about remaining charts. 

Also, for those charts that are converted into Markdown please remove the uploaded picture files from `img` folder.

File: docs/.vitepress/config.mjs (hunk 6 above)
Comment (Aish1990) [reply docs/.vitepress/config.mjs]: image

Getting above error without this change...

Comment (dmohns) [reply docs/.vitepress/config.mjs]: This change was introduced in version `1.23` of `vitepress-sidebar`, see https://vitepress-sidebar.jooy2.com/changelog#_1-23-0-2024-05-13

But we are using `1.22`. Can you please check your local setup? Something seems wrong. If you want you can send a separate PR to update to the latest release of `vitepress-sidebar`

**Review by dmohns — APPROVED** [review]

**Pull request merged**